lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -115,7 +116,7 @@ def eval_block
 
         def verify_message
           return true if @expected_message.nil?
-          values_match?(@expected_message, @actual_error.message)
+          values_match?(@expected_message, @actual_error.message.to_s)
```

Comment:
I've fixed the custom matcher, and let my raise_error changes, I'm not sure if adding a to_s where you suggest would also fix the issue, as the nested exception is a bit weird anyway, so I'm happy with this as is now